### DISCUSSION TITLE
How do I disable oh-my-posh temporarely

### DISCUSSION THREAD

**Discussion by AndrewSav** [JanDeDobbeleer/oh-my-posh#3672] [General]
When I set `GODEBUG=inittrace=1` I'm starting getting traces from oh-my-posh. I'm interested in traces from my stuff instead, so I would like to temporarily disable oh-my-posh and then re-enable it afterwards. How do I do that?

**Comment by JanDeDobbeleer** [comment]
Not possible as far as I can see.

**Reply by AndrewSav** [reply]
What's not possible? Separating traces? Sure it's not possible. But switching oh-my-posh off entirely temporarily should be possible? What is the technical limitation? @JanDeDobbeleer

**Reply by JanDeDobbeleer** [reply]
Yes, excluding. Disabling would be by commenting out the init statement.

**Reply by AndrewSav** [reply]
In the profile? That makes sense. I was hoping for a command to swich off and on dynamically, e.g. `oh-my-posh off pwsh | Invoke-Expression` etc... this way you do not need to restart the terminal. I do not keep session history between sessions for security reason, so I cannot integrate restarting terminal often into my workflow ;(

**Reply by JanDeDobbeleer** [reply]
it overrides the prompt function so you can override that one yourself if you want to.

**Reply by GameLord2011** [reply]
@JanDeDobbeleer maybe add a command to turn it off for the session?

**Reply by JanDeDobbeleer** [reply]
@gameking2011 not easy, nor in the scope of the project. It overrides the prompt functions which is mostly complicated logic.

**Comment by mrwensveen** [comment]
You can start a new shell without loading your profile with `pwsh --NoProfile` or `pwsh -nop`. Another way would be to start a new shell with an environment variable that you would then test in your profile whether or not to start oh-my-posh: `Start-Process -Environment @{ DISABLE_OMP = '1' } pwsh`. Inside `$PROFILE`:
```powershell
# ..
if (-not $env:DISABLE_OMP) {
  oh-my-posh init etc etc
}
```
